syntax = "proto3";

package user;

import "google/protobuf/empty.proto";

service UserService {
    rpc CreateUser (CreateUserRequest) returns (CreateUserResponse);
    rpc UpdateUser (UpdateUserRequest) returns (UpdateUserResponse);
    rpc HealthCheck (google.protobuf.Empty) returns (HealthCheckResponse);
}

message CreateUserRequest {
    string id = 1;
    optional string first_name = 2;
    optional string last_name = 3;
}

message CreateUserResponse {
    string id = 1;
    optional string first_name = 2;
    optional string last_name = 3;
}

message UpdateUserRequest {
    string id = 1;
    optional string first_name = 2;
    optional string last_name = 3;
}

message UpdateUserResponse {
    string id = 1;
    optional string first_name = 2;
    optional string last_name = 3;
}

message HealthCheckResponse {
    string status = 1;
    map<string, string> checks = 2;
}